### ISSUE TITLE
concat - RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by ethul** [purescript/purescript-arrays#41]
On my machine I am getting a `RangeError` when using `concat` on an array with 150,000 elements; e.g.,

```javascript
function concat (xss) {
  var result = [];
  for (var i = 0, l = xss.length; i < l; i++) {
    result.push.apply(result, xss[i]);
  }
  return result;
}

var as = new Array(150000);
var bs = concat([as]);
```

```bash
$ node index.js

/index.js:4
    result.push.apply(result, xss[i]);
                ^
RangeError: Maximum call stack size exceeded
```

**Comment by ethul** [comment]
Just saw that PR #24 also fixes this.

**Comment by garyb** [comment]
Resolved by #42, even though it wasn't actually merged :smile:
